sanda10/favorite-cities
# Add User Location to Favorites and Merge with API Favorites

Description

This PR enhances the FavoritesPage to include user location detection and merging of favorite destinations from the API with locally stored favorites. Key updates are as follows:

1. User Location Detection:

Uses the navigator.geolocation API to detect the user's current location (latitude and longitude).
The user's location is stored in localStorage and displayed on the Favorites page as a special "User Location" entry.
Handles errors gracefully if geolocation is denied or unsupported.

2.  Favorites Merging:

Combines data from the API (/api/favorites) with the user's stored favorites from localStorage.
Ensures no duplicates are displayed and allows for smooth updates when new locations are added.

3.  UI Enhancements:

Displays the user's location along with API-provided favorites.
Shows city, country, latitude, and longitude where available.

4.  Error Handling:

Added error handling for geolocation failures and API fetch issues to ensure a better user experience.



How to Test:

1.  User Location Test

Allow location access when prompted.
Check if the current location appears in the favorites list with latitude and longitude details.

2.  API Favorites Test

Ensure favorite destinations from /api/favorites are listed on the page.

3.  LocalStorage Persistence Test

Refresh the page and confirm the user location is still displayed.


Review Instructions:

Verify the logic for location fetching and merging of data from localStorage and API.
Ensure the UI updates properly after geolocation permissions are granted or denied.
Confirm that no duplicates are displayed in the list of favorites.
